### DISCUSSION TITLE
Make Views/Templates accessible for modifying/overriding (if they aren't)

### DISCUSSION THREAD

**Discussion by mgratch** [pods-framework/pods#6686] [Ideas]
I would like a way to modify relationship field HTML structure.  Currently any modifications to the fields via JS is immediately removed upon render(). Or am I wrong and this is already possible? Perhaps @pglewis has some insight?

**Comment by pglewis** [comment]
Give me a basic run-down of the use case(s) you have in mind when you get the chance.  

Our DFV based fields do not build markup on the server side and instead deliver a data script that describes the field and data.  The field is then constructed dynamically after page load, converting the data scripts into actual markup.  

There is a lot of room for flexibility with how we're doing things now.  We can support some really neat dynamic things but real world use cases will dictate which ones we should focus on.

**Comment by mgratch** [comment]
First I mentioned this to @sc0ttkclark: This implementation for related object is amazing. By far one of, if not the best I have experienced.  I love what the team has been doing! 

I spent a little bit of time review the DFV code and I wasn't sure if it was possible to modify the template dynamically, it doesn't appear to be accessible via the `PodsDFV` object.

I was trying to change the output of the list-item.html to include additional input fields.

**Here is a use case:**

1. I have a custom post type: "Sliders"
2. When editing a Slider I have a multiple pick field for another CPT "Slides".  I want to choose my slides and then on each slide enter a duration and choose an animation type.
3. Now I know that this is group field and will not be available in pods in the near future.  Therefore I decide "Hey the list field output is simply using jQuery Sortable... I can throw some JS at that and insert my form fields."  

But as I mentioned above, I learned sadly that when render fires it removes everything, which makes sense of course after reviewing how everything was rendered.

Anyhow I am not sure what is the correct answer here, but it seemed worth bringing up to see if I was missing something.

**Comment by pglewis** [comment]
My thinking is that providing a facility for you to inject a custom view might be the most flexible thing.  You could potentially take the template we use for List View as a starting point and make a modified one.  We currently just use underscore templates but that's not set in stone (#4907).  

I have sat down and really thought out the best way to do that or the potential pitfalls, DFV is only in its infancy and developer needs will drive how the API evolves, I think.  I'd love to see something like this available to folks doing advanced stuff, just need the demand for it.

**Comment by quasel** [comment]
any milestone for it @pglewis ? or just future ;)

**Comment by pglewis** [comment]
This is vaguely tied-in with several other recent things under discussion involving DFV fields, including @JoryHogeveen, @senabean33, @sandbird2.  

Some of them are on the PHP side, some on the JS side, some a little bit of both.  At the worst I'd like to address as many of these as possible during 2.8 as we expand DFV to all the fields.

**Comment by pglewis** [comment]
Sticking it on the maintenance release track in case its something that can make it in before 2.8, but with the understanding that it might end up moving to 2.8.
